codec-http2/src/main/java/io/netty/handler/codec/http2/Http2CodecUtil.java

```diff
@@ -272,10 +272,10 @@ private boolean allowNotificationEvent() {
         public ChannelPromise setSuccess(Void result) {
             if (allowNotificationEvent()) {
                 ++successfulCount;
-                if (successfulCount == expectedCount && doneAllocating) {
-                    promise.setSuccess(result);
-                    return super.setSuccess(result);
-                }
+            }
+            if (successfulCount == expectedCount && doneAllocating && !isDone()) {
```

Comment:
Can this PR just be reduced to having `return super.setSuccess();` change to `return super.setSuccess(null);` in `doneAllocatingPromises`?